apps/dh/api-dh/source/DataHub.WebApi/Utilities/DocumentFormatter.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Text.Json;
+using System.Xml.Linq;
+
+namespace Energinet.DataHub.WebApi.Utilities;
+
+public static class DocumentFormatter
+{
+    private const int MaxDocumentSizeForFormatting = 100 * 1024 * 1024; // 100 MB - safely handles 50MB documents
+
+    private static readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        WriteIndented = true,
+        Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+        DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.Never,
+        PropertyNamingPolicy = null,
+    };
+
+    public static string FormatDocumentIfNeeded(string document)
+    {
+        if (string.IsNullOrWhiteSpace(document))
+        {
+            return document;
+        }
+
+        if (document.Length > MaxDocumentSizeForFormatting)
+        {
+            return document;
+        }
+
+        if (TryFormatJson(document, out var formattedJson))
+        {
+            return formattedJson;
+        }
+
+        if (TryFormatXml(document, out var formattedXml))
+        {
+            return formattedXml;
+        }
+
+        return document;
+    }
+
+    private static bool IsAlreadyFormatted(string input)
```

Comment:
question: is this necessary? i assume it is to save time/processing, but in the near future, it will only be true for old messages. perhaps it is better to trade some CPU cycles for reduced complexity 🙂

Reply:
Should be removed now